tests/ga/test_extension.py

```diff
@@ -2322,324 +2324,320 @@ def test_it_should_parse_boolean_values_encoded_as_strings(self):
+                exthandlers_handler.run()
+            
+                return invocation_record
 
-        with patch.object(CGroupConfigurator.get_instance(), "start_extension_command",
-                          side_effect=['ok', 'ok', ExtensionError('uninstall Failed'), 'ok']) as patch_start_cmd:
 
-            ExtHandlersHandler._update_extension_handler_and_return_if_failed(old_handler_i, new_handler_i)
+    def test_non_enabled_ext_should_not_be_disabled_at_ver_update(self):
 
-            args, kwargs = patch_start_cmd.call_args
+        first_ext = extension_emulator(enable_action=Actions.fail_action)
+        second_ext = extension_emulator(version="1.1.0")
+        
+        invocation_record = TestExtensionUpdateOnFailure._do_upgrade_scenario(first_ext, second_ext)
+
+        invocation_record.compare(
+            (second_ext, ExtensionCommandNames.UPDATE),
+            (first_ext, ExtensionCommandNames.UNINSTALL),
+            (second_ext, ExtensionCommandNames.INSTALL),
+            (second_ext, ExtensionCommandNames.ENABLE)
+        )
+
+    def test_disable_failed_env_variable_should_be_set_for_update_cmd_when_continue_on_update_failure_is_true(self):
+        exit_code, disable_action = Actions.generate_unique_fail()
+
+        first_ext = extension_emulator(disable_action=disable_action)
+        second_ext = extension_emulator(version="1.1.0", continue_on_update_failure=True)
+
+        invocation_record = TestExtensionUpdateOnFailure._do_upgrade_scenario(first_ext, second_ext)
+
+        invocation_record.compare(
+            (first_ext, ExtensionCommandNames.DISABLE),
+            (second_ext, ExtensionCommandNames.UPDATE),
+            (first_ext, ExtensionCommandNames.UNINSTALL),
+            (second_ext, ExtensionCommandNames.INSTALL),
+            (second_ext, ExtensionCommandNames.ENABLE)
+        )
+
+        _, kwargs = second_ext.actions[ExtensionCommandNames.UPDATE].call_args
+
+        self.assertEqual(kwargs["env"][ExtCommandEnvVariable.DisableReturnCode], exit_code,
+            "DisableAction's return code should be in updateAction's env.")
+
+
+    def test_uninstall_failed_env_variable_should_set_for_install_when_continue_on_update_failure_is_true(self):
+        exit_code, uninstall_action = Actions.generate_unique_fail()
+
+        first_ext = extension_emulator(uninstall_action=uninstall_action)
+        second_ext = extension_emulator(version="1.1.0", continue_on_update_failure=True)
+
+        invocation_record = TestExtensionUpdateOnFailure._do_upgrade_scenario(first_ext, second_ext)
+
+        invocation_record.compare(
+            (first_ext, ExtensionCommandNames.DISABLE),
+            (second_ext, ExtensionCommandNames.UPDATE),
+            (first_ext, ExtensionCommandNames.UNINSTALL),
+            (second_ext, ExtensionCommandNames.INSTALL),
+            (second_ext, ExtensionCommandNames.ENABLE)
+        )
+
+        _, kwargs = second_ext.actions[ExtensionCommandNames.INSTALL].call_args
```

Comment:
A suggestion, you could also pass a lambda as an action to verify if a certain env variable is set. That way you will have one less state to maintain in your Mocks.

For eg: 
```python
install_action = lambda *args, *kwargs: return 0 if "Uninstall_failed" in args['env'] else 1 
second_ext = extension_emulator(version="1.1.0", continue_on_update_failure=True, install_action=install_action)
```

Something like this (this exact example might not work exactly as it, just wanted to illustrate my point)

Reply:
This could maybe work. I'd love to sync with you offline to discuss this further-- I'm mostly concerned that this method would prematurely exit the `run` call, and the ramifications that might have in our coverage...

Reply:
Sure let's sync about it. I dont understand exactly what you mean by premature exit, Maybe you could explain more over in our sync

Reply:
Not sure if I follow, did you guys sync about this?

Reply:
We did! we ultimately decided that both ways of doing things (i.e. the way in the code, and the way @larohra suggested) are functionally the same for this particular use case, but that there could conceivably be a scenario which would better fit the suggested code.